### PR TITLE
Add support of multi valued types to ConfigProperties

### LINKED ISSUE #641: Add support of multi valued types to ConfigProperties
So far when we use the standard annotation `ConfigProperties`, we only have access to basic types, there is no way to use a `Set`, a `List` or an array because if we do so, we end up with an error of type:

```
java.lang.reflect.GenericSignatureFormatError: Signature Parse error: Expected Field Type Signature
	Remaining input: ;>;
	at java.base/sun.reflect.generics.parser.SignatureParser.error(SignatureParser.java:124)
	at java.base/sun.reflect.generics.parser.SignatureParser.parseFieldTypeSignature(SignatureParser.java:291)
	at java.base/sun.reflect.generics.parser.SignatureParser.parseFieldTypeSignature(SignatureParser.java:277)
	at java.base/sun.reflect.generics.parser.SignatureParser.parseTypeArgument(SignatureParser.java:420)
	at java.base/sun.reflect.generics.parser.SignatureParser.parseTypeArguments(SignatureParser.java:383)
	at java.base/sun.reflect.generics.parser.SignatureParser.parsePackageNameAndSimpleClassTypeSignature(SignatureParser.java:335)
	at java.base/sun.reflect.generics.parser.SignatureParser.parseClassTypeSignature(SignatureParser.java:304)
	at java.base/sun.reflect.generics.parser.SignatureParser.parseFieldTypeSignature(SignatureParser.java:283)
	at java.base/sun.reflect.generics.parser.SignatureParser.parseFieldTypeSignature(SignatureParser.java:277)
	at java.base/sun.reflect.generics.parser.SignatureParser.parseTypeSignature(SignatureParser.java:469)
	at java.base/sun.reflect.generics.parser.SignatureParser.parseReturnType(SignatureParser.java:611)
	at java.base/sun.reflect.generics.parser.SignatureParser.parseMethodTypeSignature(SignatureParser.java:561)
	at java.base/sun.reflect.generics.parser.SignatureParser.parseMethodSig(SignatureParser.java:171)
	at java.base/sun.reflect.generics.repository.ConstructorRepository.parse(ConstructorRepository.java:58)
	at java.base/sun.reflect.generics.repository.ConstructorRepository.parse(ConstructorRepository.java:43)
	at java.base/sun.reflect.generics.repository.AbstractRepository.<init>(AbstractRepository.java:74)
	at java.base/sun.reflect.generics.repository.GenericDeclRepository.<init>(GenericDeclRepository.java:49)
	at java.base/sun.reflect.generics.repository.ConstructorRepository.<init>(ConstructorRepository.java:54)
	at java.base/sun.reflect.generics.repository.MethodRepository.<init>(MethodRepository.java:47)
	at java.base/sun.reflect.generics.repository.MethodRepository.make(MethodRepository.java:60)
	at java.base/java.lang.reflect.Method.getGenericInfo(Method.java:109)
	at java.base/java.lang.reflect.Method.getGenericReturnType(Method.java:292)
	at io.smallrye.config.ConfigMappingInterface.getProperties(ConfigMappingInterface.java:602)
	at io.smallrye.config.ConfigMappingInterface.getProperties(ConfigMappingInterface.java:603)
	at io.smallrye.config.ConfigMappingInterface.getProperties(ConfigMappingInterface.java:603)
	at io.smallrye.config.ConfigMappingInterface.getProperties(ConfigMappingInterface.java:603)
	at io.smallrye.config.ConfigMappingInterface.getProperties(ConfigMappingInterface.java:603)
	at io.smallrye.config.ConfigMappingInterface.getProperties(ConfigMappingInterface.java:603)
	at io.smallrye.config.ConfigMappingInterface.getProperties(ConfigMappingInterface.java:594)
```

### PR DESCRIPTION
**Pull request by essobedo**
fixes #641

## Motivation

There is no way to use a `Set`, a `List` or an array as type of a class annotated with `ConfigProperties`.

## Modifications:

* Improve the method allowing to generate the signature in "byte-code format" to support properly collections and arrays.
* Force the language at surefire level to ensure that `ValidateConfigTest` will pass whatever the default system language (not related to the issue)

### CODE REVIEW

**Comment by essobedo** [comment]
Hi @radcortez, I know that you are not a big fan of `ConfigProperties` 😃  but still it would be nice to have it 🙏  , don't you agree? Maybe I can propose it to the expert group directly?

File: validator/pom.xml [hunk 1]
```diff
@@ -65,4 +65,17 @@
             <scope>test</scope>
         </dependency>
     </dependencies>
+    <build>
```
Comment (radcortez) [inline validator/pom.xml:68]: Do we need this? It doesn't seem related.

**Review by radcortez — CHANGES_REQUESTED** [review]

**Comment by radcortez** [comment]
> Hi @radcortez, I know that you are not a big fan of `ConfigProperties` 

Only because they are classes, that is all. If it was an interface it would be fine :)

File: validator/pom.xml (hunk 1 above)
Comment (essobedo) [reply validator/pom.xml:68]: Indeed, it's not related, it is to fix `ValidateConfigTest` on non english system. Do you want me to remove it?

**Review by radcortez — APPROVED** [review]

**Comment by radcortez** [comment]
Thanks for the PR :)
